src/many/src/server/module/_9_account.rs

```diff
@@ -0,0 +1,563 @@
+            Self(AccountMap::new(
+                Identity::from_bytes(
+                    &hex::decode("0102030405060708090A0B0C0D0E0F101112131415161718191A1B1C1D")
+                        .unwrap(),
+                )
+                .unwrap(),
+            ))
+        }
+    }
+
+    impl super::AccountModuleBackend for AccountImpl {
+        fn create(
+            &mut self,
+            sender: &Identity,
+            args: CreateArgs,
+        ) -> Result<CreateReturn, ManyError> {
+            let (id, _) = self.0.insert(Account::create(sender, args))?;
+            Ok(CreateReturn { id })
+        }
+
+        fn set_description(
+            &mut self,
+            _sender: &Identity,
+            args: SetDescriptionArgs,
+        ) -> Result<EmptyReturn, ManyError> {
+            let mut account = self
+                .0
+                .get_mut(&args.id)
+                .ok_or_else(|| errors::unknown_account(args.id.to_string()))?;
+
+            account.description = Some(args.description);
+            Ok(EmptyReturn)
+        }
+
+        fn list_roles(
+            &self,
+            _sender: &Identity,
+            args: ListRolesArgs,
+        ) -> Result<ListRolesReturn, ManyError> {
+            let _ = self
+                .0
+                .get(&args.account)
+                .ok_or_else(|| errors::unknown_account(args.account.to_string()))?;
+
+            Ok(ListRolesReturn {
+                roles: BTreeSet::from_iter(
+                    vec!["owner".to_string(), "other-role".to_string()].into_iter(),
+                ),
+            })
+        }
+
+        fn get_roles(
+            &self,
+            _sender: &Identity,
+            args: GetRolesArgs,
+        ) -> Result<GetRolesReturn, ManyError> {
+            let account = self
+                .0
+                .get(&args.account)
+                .ok_or_else(|| errors::unknown_account(args.account.to_string()))?;
+
+            Ok(GetRolesReturn {
+                roles: account.roles.clone(),
+            })
+        }
+
+        fn add_roles(
+            &mut self,
+            _sender: &Identity,
+            args: AddRolesArgs,
+        ) -> Result<EmptyReturn, ManyError> {
+            let account = self
+                .0
+                .get_mut(&args.account)
+                .ok_or_else(|| errors::unknown_account(args.account.to_string()))?;
+
+            for (k, mut v) in args.roles {
+                let roles = account.roles.entry(k).or_default();
+                roles.append(&mut v);
+            }
+            Ok(EmptyReturn)
+        }
+
+        fn remove_roles(
+            &mut self,
+            _sender: &Identity,
+            args: RemoveRolesArgs,
+        ) -> Result<EmptyReturn, ManyError> {
+            let account = self
+                .0
+                .get_mut(&args.account)
+                .ok_or_else(|| errors::unknown_account(args.account.to_string()))?;
+
+            for (k, v) in args.roles {
```

Comment:
nit: Remove entry from `account.roles` if value is empty after role removal?

Reply:
If we ever move logic to this crate we should, but this is test logic.